server/sonar-server/src/main/java/org/sonar/server/measure/persistence/MeasureDao.java

```diff
@@ -55,6 +56,10 @@ public void insert(DbSession session, MeasureDto measureDto) {
     mapper(session).insert(measureDto);
   }
 
+  public List<String> selectMetricKeysForSnapshot(DbSession session, Long snapshotId) {
```

Comment:
snapshotId can't be null, so I would enforce that by using a primitive type "long"